packages/vaadin-overlay/src/vaadin-overlay-position-mixin.js

```diff
@@ -67,125 +77,130 @@ export const _PositionMixin = (superClass) =>
     }
     static get observers() {
       return [
-        `__positionSettingsChanged(positionTarget, horizontalAlign, verticalAlign,
-      noHorizontalOverlap, noVerticalOverlap)`
+        `__positionSettingsChanged(positionTarget, horizontalAlign, verticalAlign, noHorizontalOverlap, noVerticalOverlap)`,
+        `__overlayOpenedChanged(opened)`
       ];
     }
 
     constructor() {
       super();
 
-      const boundUpdatePosition = this._updatePosition.bind(this);
-
-      this.addEventListener('opened-changed', (e) => {
-        const func = e.detail.value ? 'addEventListener' : 'removeEventListener';
-        window[func]('scroll', boundUpdatePosition);
-        window[func]('resize', boundUpdatePosition);
+      this.__boundUpdatePosition = this._updatePosition.bind(this);
+    }
 
-        if (e.detail.value) {
-          this._updatePosition();
+    __overlayOpenedChanged(opened) {
+      // Toggle the event listeners that cause the overlay to update its position
+      ['scroll', 'resize'].forEach((eventName) => {
+        if (opened) {
+          window.addEventListener(eventName, this.__boundUpdatePosition);
+        } else {
+          window.removeEventListener(eventName, this.__boundUpdatePosition);
         }
       });
-    }
 
-    ready() {
-      super.ready();
+      if (opened) {
+        const computedStyle = getComputedStyle(this);
+        if (!this.__margins) {
+          this.__margins = {};
+          ['top', 'bottom', 'left', 'right'].forEach((propName) => {
+            this.__margins[propName] = parseInt(computedStyle[propName], 10);
+          });
+        }
+        this.__isRTL = computedStyle.direction === 'rtl';
 
-      console.warn('PositionMixin is not considered stable and might change any time');
+        this._updatePosition();
+        requestAnimationFrame(() => this._updatePosition());
```

Comment:
This probably needs the same comment as we had in combo-box (about virtual keyboard on mobile).